### PR TITLE
Add more lookup checks when creating txn

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description

Add stricter check on lookup level to reject transaction.

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libServer/LookupServer.cpp [hunk 1]
```diff
@@ -341,6 +341,29 @@ bool LookupServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
...
+    throw JsonRpcException(
+        RPC_INVALID_PARAMETER,
+        "Gas limit lower than required for creating contract, Minimum: " +
+            to_string(CONTRACT_CREATE_GAS));
+  }
+
+  if (sender->GetNonce() >= tx.GetNonce()) {
+    throw JsonRpcException(
+        RPC_INVALID_PARAMETER,
+        "Nonce too low, Current Nonce: " + to_string(sender->GetNonce()));
```
Comment (ansnunez) [inline src/libServer/LookupServer.cpp]:
```suggestion
        "Nonce (" + to_string(tx.GetNonce()) + ") lower than current (" + to_string(sender->GetNonce()) + ")");
```

File: src/libServer/LookupServer.cpp [hunk 2]
```diff
@@ -341,6 +341,29 @@ bool LookupServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
...
+
+  if (type == Transaction::ContractType::CONTRACT_CALL &&
+      (tx.GetGasLimit() < CONTRACT_INVOKE_GAS)) {
+    throw JsonRpcException(
+        RPC_INVALID_PARAMETER,
+        "Gas limit lower than required for invoking contract, Minimum: " +
+            to_string(CONTRACT_INVOKE_GAS));
+  }
+
+  if (type == Transaction::ContractType::CONTRACT_CREATION &&
```
Comment (ansnunez) [inline src/libServer/LookupServer.cpp]:
```suggestion
  else if (type == Transaction::ContractType::CONTRACT_CREATION &&
```

File: src/libServer/LookupServer.cpp [hunk 3]
```diff
@@ -341,6 +341,30 @@ bool LookupServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
     throw JsonRpcException(RPC_INVALID_ADDRESS_OR_KEY,
                            "The sender of the txn has no balance");
   }
+  const auto type = Transaction::GetTransactionType(tx);
+
+  if (type == Transaction::ContractType::CONTRACT_CALL &&
+      (tx.GetGasLimit() < CONTRACT_INVOKE_GAS)) {
```
Comment (kaikawaliu) [inline src/libServer/LookupServer.cpp]: std::max(CONTRACT_INVOKE_GAS, (unsigned int)(transaction.GetData().size()))

File: src/libServer/LookupServer.cpp [hunk 4]
```diff
@@ -341,6 +341,30 @@ bool LookupServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
...
+    throw JsonRpcException(RPC_INVALID_PARAMETER,
+                           "Gas limit (" + to_string(tx.GetGasLimit()) +
+                               ") lower than minimum for invoking contract (" +
+                               to_string(CONTRACT_INVOKE_GAS) + ")");
+  }
+
+  else if (type == Transaction::ContractType::CONTRACT_CREATION &&
+           (tx.GetGasLimit() < CONTRACT_CREATE_GAS)) {
```
Comment (kaikawaliu) [inline src/libServer/LookupServer.cpp]: std::max(CONTRACT_CREATE_GAS, (unsigned int)(transaction.GetCode().size() + transaction.GetData().size()))

**Comment by kaikawaliu** [comment]
> Can just put the check LookupServer::CreateTransaction to avoid reconstruct the switch conditions.

This one is fine

**Pull request merged**